src/Framework/Attributes/RequiresEnvironmentVariable.php

```diff
@@ -0,0 +1,40 @@
+<?php declare(strict_types=1);
+/*
+ * This file is part of PHPUnit.
+ *
+ * (c) Sebastian Bergmann <sebastian@phpunit.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace PHPUnit\Framework\Attributes;
+
+use Attribute;
+
+/**
+ * @immutable
+ *
+ * @no-named-arguments Parameter names are not covered by the backward compatibility promise for PHPUnit
+ */
+#[Attribute(Attribute::TARGET_CLASS | Attribute::TARGET_METHOD | Attribute::IS_REPEATABLE)]
+final readonly class RequiresEnvironmentVariable
+{
+    private string $environmentVariableName;
+    private null|bool|int|string $value;
```

Comment:
Environment variable can be strictly only string.

What is the meaning of `null`, `bool`, `int`?

Can I test environment variable presence/absence /wo value?

Reply:
Based on https://github.com/sebastianbergmann/phpunit/pull/6074/files#diff-8f0e8d461aea381bf08d49440fbc5da892d38eb782518cd44bfe1dddce7638f4R119 it seems `bool` and `int` should be removed and `null` is used for testing if the variable is present /wo specific value.

Reply:
> Can I test environment variable presence/absence /wo specific value?

you can only test environment variable presence or with a specific value

> it seems bool and int should be removed and as never possible 

this was more or less my concern [here](https://github.com/sebastianbergmann/phpunit/pull/6074#discussion_r1875030847). I'm also wondering if we should only check the env var in `$_ENV` , or use `getenv()` or check at least one of them

Reply:
any thoughts @sebastianbergmann? I can provide a patch to my PR

Reply:
> this was more or less my concern [here](https://github.com/sebastianbergmann/phpunit/pull/6074#discussion_r1875030847). I'm also wondering if we should only check the env var in `$_ENV` , or use `getenv()` or check at least one of them

Please test the behaviour with E2E test and some tests in separate process. The separate process must see the same environment variables in both linux and Windows.

Reply:
AFAICS, an environment variable can only be `string`. And just checking whether an environment variable is set or not without checking for an expected value does not make much sense to me.

Reply:
ok, let me fix this

Reply:
see https://github.com/php-lock/lock/blob/2.3.0/tests/mutex/MutexConcurrencyTest.php#L26 for legit usecases of "variable is set"

Reply:
yeah, actually in https://github.com/zenstruck/foundry we also have legit usecases for only "variable is set" :sweat_smile:

Reply:
Fair enough.

Reply:
do you think in the case of "no value", I check if the env var exist, or should I check it is ok-ish? empty string and `'0'` would not pass

Reply:
In my opinion:

- If an environment variable is tested for a presence (/wo value defined in the Attribute), I would expect `'0'` to be consideted as set.

- For binary `'1'`/`'0'` detection I would expect user to write `#[RequiresEnvironmentVariable('foo', '1')]` explicitly.

- `''` should be considered as unset as in many operating systems `FOO=` unsets the variable. Also, in CI parametrization/matrix empty value might be set, but with "no variable" meaning.

Reply:
> '' should be considered as unset as in many operating systems FOO= unsets the variable

yeah, good point